jcl/src/openj9.sharedclasses/share/classes/com/ibm/oti/shared/SharedClassAbstractHelper.java

```diff
@@ -94,17 +94,35 @@ private String recursiveJarTrim(String jarName) {
 		if (jarName==null)
 			return null;
 
-		int lastIndexBang = jarName.lastIndexOf("!/"); //$NON-NLS-1$
 		boolean startsWithJar = jarName.startsWith("jar:"); //$NON-NLS-1$
+		boolean endsWithBang = jarName.endsWith("!/");//$NON-NLS-1$
 		int subStringStart = startsWithJar ? 4 : 0;
-		int subStringEnd = lastIndexBang>=0 ? lastIndexBang : jarName.length();
+		int len = jarName.length();
+		int subStringEnd = endsWithBang ? len -2 : len;
 
-		if (!startsWithJar && lastIndexBang==-1) {
+		if (!startsWithJar && !endsWithBang) {
 			return jarName;
 		} else {
 			return recursiveJarTrim(jarName.substring(subStringStart, subStringEnd));
 		}
 	}
+	
+	private static URL getURLToCheck(URL url) {
+		String pathString = url.toString();
+		int indexBang = pathString.indexOf(".jar!/"); //$NON-NLS-1$
```

Comment:
The jar file doesn't need to end with ".jar". In theory it could be anything, and it would be more robust not to add any restriction, although the native code seems to restrict it to ".jar" and ".zip".

Reply:
I can change it to check "!/" only